codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/error/ErrorGenerator.kt

```diff
@@ -85,6 +85,12 @@ class ErrorGenerator(
                 """
                 /// Returns the error message.
                 pub fn message(&self) -> Option<&str> { $message }
+
+                ##[doc(hidden)]
+                /// Returns the error name.
+                pub fn name(&self) -> String {
+                    String::from("${shape.id.name}")
+                }
```

Comment:
I have added this `name()` method which returns the error name as a String. The server is using this to store any modeled error name inside `http::Extensions` to help with metrics reporting.

This is the only change to the codegen and from what I can see in the errorTrait spec, it shouldn't impact anything. Let's see what @awslabs/rust-sdk-owners think about this and what the CI will say 😄

Reply:
Also I was thinking that this can probably become

```rust
    ##[doc(hidden)]
    /// Returns the error name.
    pub fn name(&self) -> &'static str {
        ${shape.id.name.dq()}
    }
```

Thoughts?

Reply:
👍🏼